dbtsl/api/graphql/client/base.py

```diff
@@ -28,14 +29,22 @@ class BaseGraphQLClient(Generic[TTransport, TSession]):
 
     PROTOCOL = GraphQLProtocol
     DEFAULT_URL_FORMAT = env.DEFAULT_GRAPHQL_URL_FORMAT
+    DEFAULT_TIMEOUT = TimeoutOptions(
+        total_timeout=30,
+        # Ideally, connect timeouts are a little more than a multiple of 3, which
+        # is the default packet retransmission window for TCP
+        # See: https://datatracker.ietf.org/doc/html/rfc2988
+        connect_timeout=4,
+        execute_timeout=10,
```

Comment:
similarly... I know there's compile requests that take upwards of 30s, maybe we should also look at some DD stats to figure out an optimal value here